File: demo-android/app/src/androidTest/java/com/example/demo/e2e/test/delete/DeleteTaskTest.kt
package com.example.demo.e2e.test.delete

import com.example.demo.e2e.context.TaskTestContext
import com.example.demo.e2e.data.AllureEpic.TASK_MANAGEMENT
import com.example.demo.e2e.data.TaskResponse
import com.example.demo.e2e.test.base.MockedBackendTestBase
import io.qameta.allure.kotlin.Epic
import io.qameta.allure.kotlin.Feature
import io.qameta.allure.kotlin.TmsLink
import org.junit.Before
import org.junit.Test

@Epic(TASK_MANAGEMENT)
@Feature("Delete task")
@TmsLink("99")
@TmsLink("115")
class DeleteTaskTest : MockedBackendTestBase() {
    private lateinit var context: TaskTestContext
    private lateinit var response: TaskResponse

    @Before
    fun setup() {
        context = TaskTestContext()
        response = context.createTaskResponse()

        support().mock.api().getTasks(response.toTask())
        support().mock.api().deleteTask()
    }

    @Test
    fun shouldDeleteTask() {
        // Given
        steps.navigation.openMainPage()
        steps.navigation.refreshMainPage()

        // When
        steps.tasks.deleteTask(response.title)

        // Then
        validate.tasks.hasNoTask(response.title)
    }
}

File: demo-ios/DemoUITests/Test/Delete/DeleteTaskTest.swift
import XCTest

final class DeleteTaskTest: MockedBackendTestBase, AllureAnnotated {
    static var allureEpic: String { AllureEpic.taskManagement }
    static var allureFeature: String { "Delete task" }
    static var allureTmsLinks: [String] { ["99", "115"] }

    private var context: TaskTestContext!
    private var response: TaskResponse!

    override func stubApis() throws {
        context = TaskTestContext()
        response = context.createTaskResponse()

        try support.mock.api()
            .getTasks(response)
            .deleteTask()
    }

    func testShouldRemoveTaskFromListWhenDeleteSucceeds() {
        // Given
        steps.navigation.openMainPage()

        // When
        steps.tasks.deleteTask(response.title)

        // Then
        validate.tasks.hasNoTask(response.title)
    }
}
